tests/foreman/cli/test_satellitesync.py

```diff
@@ -16,95 +16,121 @@
+    """Check settings and set download policy for export.  Reset to original state after import"""
+    download_policy_value = Settings.info({'name': 'default_download_policy'})['value']
+    rh_download_policy_value = Settings.info({'name': 'default_redhat_download_policy'})['value']
+    disconnected_mode_value = Settings.info({'name': 'content_disconnected'})['value']
+    assert download_policy_value == 'immediate'
+    assert rh_download_policy_value == 'on_demand'
+    assert disconnected_mode_value == 'false'
+    Settings.set({'name': 'default_download_policy', 'value': 'immediate'})
+    yield
+    Settings.set({'name': 'default_download_policy', 'value': download_policy_value})
+    Settings.set({'name': 'default_redhat_download_policy', 'value': rh_download_policy_value})
+    Settings.set({'name': 'content_disconnected', 'value': disconnected_mode_value})
 
 
 @pytest.mark.run_in_one_thread
 class TestRepositoryExport:
     """Tests for exporting a repository via CLI"""
 
     @pytest.mark.tier3
-    def test_positive_export_custom_product(self, class_export_directory, module_org):
-        """Export a repository from the custom product
+    def test_positive_export_complete_version_custom_repo(self, default_sat, module_org):
+        """Export a custom repository via complete version
 
         :id: 9c855866-b9b1-4e32-b3eb-7342fdaa7116
 
-        :expectedresults: Repository was successfully exported, rpm files are
+        :expectedresults: Repository was successfully exported, exported files are
+            present on satellite machine
+
+        :CaseLevel: System
+        """
+        # setup custom repo
+        cv_name = gen_string('alpha')
+        export_dir = '/var/lib/pulp/exports/'
+        product = make_product({'organization-id': module_org.id})
+        repo = make_repository(
+            {
+                'download-policy': 'immediate',
+                'organization-id': module_org.id,
+                'product-id': product['id'],
+            }
+        )
+        Repository.synchronize({'id': repo['id']})
+        # create cv and publish
+        cv = make_content_view({'name': cv_name, 'organization-id': module_org.id})
+        ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo['id'],
+            }
+        )
+        ContentView.publish({'id': cv['id']})
+        cv = ContentView.info({'id': cv['id']})
+        assert len(cv['versions']) == 1
+        cvv = cv['versions'][0]
+        # Verify export directory is empty
+        assert validate_filepath(default_sat, export_dir) == ''
+        # Export content view
+        ContentExport.completeVersion({'id': cvv['id'], 'organization-id': module_org.id})
+        # Verify export directory is not empty
+        assert validate_filepath(default_sat, export_dir) != ''
+        # test cleanup: remove org dir in export dir
```

Comment:
Test cleanup should be done in finalizers, not in the test function itself. Maybe the impact is low here if cleanup doesn't run, but its the same comment that I made before - if any of the statements raise exceptions or assertions fail during the test, this cleanup won't run.

Reply:
If you use the fixtures for org creation instead of creating them in the test function by calling `make_org()` (other test functions, this is already using `module_org`), you can write a fixture that yields and then does this cleanup.

`export_dir` is defined several times in this module, always to the same path. That can be defined as a capsule class attribute, or as a constant in the module - that way the cleanup function doesn't have to hard code it.